### PR TITLE
Sprint 17

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 17

### [1335: Petitions and Docket: Different inbox for external messages](https://github.com/flexion/ef-cms/issues/1335)
image

### [1491: System validates that PDF is valid PDF during upload](https://github.com/flexion/ef-cms/issues/1491)
image

### [1889: Fix respondent association](https://github.com/flexion/ef-cms/issues/1889)
image

### [1192: Calendar Clerk: Run automated scheduling of trial sessions](https://github.com/flexion/ef-cms/issues/1192)
image

### CODE REVIEW

**Comment by waldoj** [comment]
Here's the review checklist that I'll work my way through in reviewing this PR:

- [ ] begin by deleting any existing copy of the Git repository and cloning a fresh copy, running `docker system prune` to remove any cached artifacts from any prior code review, and running `./clean-up.sh` to update Node packages
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [ ] run a security audit of dependencies (`npm audit`) in `efcms-service`, `web-client`, and `shared`, to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](https://github.com/ustaxcourt/ef-cms/blob/staging/docs/CODE_REVIEW.md#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
Because GitHub lacks effective pull-request-level feedback mechanisms that must be addressed before a pull request can be merged, I'll be using `README.md` as the place to attach PR-level comments.

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ An as-yet-unnamed project by the [U.S. Tax Court](https://ustaxcourt.gov/), crea
...
...roject_badges/measure?project=ef-cms-front-end&metric=sqale_rating)](https://sonarcloud.io/dashboard?id=ef-cms-front-end)<br>[![Security Rating](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-front-end&metric=security_rating)](https://sonarcloud.io/dashboard?id=ef-cms-front-end)<br>[![Known Vulnerabilities](https://snyk.io/test/github/ustaxcourt/ef-cms/badge.svg?targetFile=web-client%2Fpackage.json)](https://snyk.io/test/github/ustaxcourt/ef-cms?targetFile=web-client%2Fpackage.json) | [![Coverage](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-shared&metric=coverage)](https://sonarcloud.io/dashboard?id=ef-cms-shared)<br>[![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-shared&metric=sqale_rating)](https://sonarcloud.io/dashboard?id=ef-cms-shared)<br>[![Security Rating](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-shared&metric=security_rating)](https://sonarcloud.io/dashboard?id=ef-cms-shared)<br>[![Known Vulnerabilities](https://snyk.io/test/github/ustaxcourt/ef-cms/badge.svg?targetFile=shared%2Fpackage.json)](https://snyk.io/test/github/ustaxcourt/ef-cms?targetFile=shared%2Fpackage.json)
```
Comment (waldoj) [inline README.md]: In the invalid-PDF error dialog ("Uh oh! An error occurred while uploading your document and it wasn’t completed."), please hyphenate `Uh oh`, so that it appears as `Uh-oh`, [per Merriam-Webster](https://www.merriam-webster.com/dictionary/uh-oh).

File: shared/src/business/entities/TrialSession.test.js [hunk 2]
```diff
@@ -82,4 +82,26 @@ describe('TrialSession entity', () => {
       expect(trialSession.generateSortKeyPrefix()).toEqual('BirminghamAL-H');
     });
   });
+
+  describe('setAsCalendared', () => {
+    it('should set trial session as calendared', () => {
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply shared/src/business/entities/TrialSession.test.js]: @kkoskelin @wbyoko could one of y'all take a look at this?

File: shared/src/business/entities/TrialSession.test.js [hunk 3]
```diff
@@ -82,4 +82,26 @@ describe('TrialSession entity', () => {
...
+        ...VALID_TRIAL_SESSION,
+        sessionType: 'Hybrid',
+      });
+      trialSession.setAsCalendared();
+      expect(trialSession.isCalendared).toEqual(true);
+    });
+  });
+
+  describe('addCaseToCalendar', () => {
+    it('should add case to calendar', () => {
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply shared/src/business/entities/TrialSession.test.js]: @kkoskelin @wbyoko could one of y'all take a look at this?

File: shared/src/business/useCases/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlInteractor.test.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+const chromiumMock = {
+  font: () => {},
+  puppeteer: {
+    launch: () => browserMock,
+  },
+};
+describe('createCourtIssuedOrderPdfFromHtmlInteractor', () => {
+  it('returns the pdf result', async () => {
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply shared/src/business/useCases/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlInteractor.test.js]: @kkoskelin @wbyoko could one of y'all take a look at this?

File: shared/src/business/useCases/trialSessions/getTrialSessionDetailsInteractor.test.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  term: 'Fall',
+  termYear: '3000',
+  trialLocation: 'Birmingham, AL',
+  trialSessionId: '208a959f-9526-4db5-b262-e58c476a4604',
+};
+
+describe('Get trial session details', () => {
+  let applicationContext;
+
+  it('throws error if user is unauthorized', async () => {
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply shared/src/business/useCases/trialSessions/getTrialSessionDetailsInteractor.test.js]: @kkoskelin @wbyoko could one of y'all take a look at this?

File: shared/src/business/useCases/trialSessions/setTrialSessionCalendarInteractor.test.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        trialSessionId: '6805d1ab-18d0-43ec-bafb-654e83405416',
+      });
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).toBeDefined();
+  });
+
+  it('should set a trial session to "calendared"', async () => {
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply shared/src/business/useCases/trialSessions/setTrialSessionCalendarInteractor.test.js]: @kkoskelin @wbyoko could one of y'all take a look at this?

File: shared/src/business/useCases/workitems/getDocumentQCServedForSectionInteractor.test.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    docketNumberSuffix: 'S',
+    document: {
+      sentBy: 'taxyaper',
+    },
+    messages: [],
+    section: 'docket',
+    sentBy: 'docketclerk',
+  };
+
+  it('throws an error if the work item was not found', async () => {
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply shared/src/business/useCases/workitems/getDocumentQCServedForSectionInteractor.test.js]: @kkoskelin @wbyoko could one of y'all take a look at this?

File: shared/src/business/useCases/workitems/getDocumentQCServedForSectionInteractor.test.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        applicationContext,
+        section: 'docket',
+      });
+    } catch (e) {
+      error = e;
+    }
+    expect(error).toBeDefined();
+  });
+
+  it('throws an error if the user does not have access to the work item', async () => {
```
[the remark this answers is not in the document]
Comment (sutt0n) [reply shared/src/business/useCases/workitems/getDocumentQCServedForSectionInteractor.test.js]: @kkoskelin @wbyoko could one of y'all take a look at this?
